flexmeasures/api/v3_0/tests/test_sensor_schedules_fresh_db.py

```diff
@@ -393,6 +392,121 @@ def test_inflexible_device_sensors_priority(
+        "site-peak-production-price": "260 EUR/MW",
+    }
+
+    sensor = (
+        Sensor.query.filter(Sensor.name == "power")
+        .join(GenericAsset, GenericAsset.id == Sensor.generic_asset_id)
+        .filter(GenericAsset.name == asset_name)
+        .one_or_none()
+    )
+
+    with app.test_client() as client:
+        trigger_schedule_response = client.post(
+            url_for("SensorAPI:trigger_schedule", id=sensor.id),
+            json=message,
+        )
+        print("Server responded with:\n%s" % trigger_schedule_response.json)
+        assert trigger_schedule_response.status_code == 200
+        job_id = trigger_schedule_response.json["schedule"]
+
+    # process the scheduling queue
+    work_on_rq(
+        app.queues["scheduling"], exc_handler=handle_scheduling_exception, max_jobs=1
+    )
+    job = Job.fetch(job_id, connection=app.queues["scheduling"].connection)
+    assert job.is_finished is True
+
+    # First, make sure the scheduler data source is now there
+    job.refresh()  # catch meta info that was added on this very instance
+    scheduler_source = get_data_source_for_job(job)
+    assert scheduler_source is not None
+
+    # try to retrieve the schedule through the /sensors/<id>/schedules/<job_id> [GET] api endpoint
+    get_schedule_response = client.get(
+        url_for("SensorAPI:get_schedule", id=sensor.id, uuid=job_id),
+        query_string={"duration": "PT48H"},
+    )
+    print("Server responded with:\n%s" % get_schedule_response.json)
+    assert get_schedule_response.status_code == 200
+    power_values = get_schedule_response.json["values"]
+    start = get_schedule_response.json["start"]
+    duration = get_schedule_response.json["duration"]
+
+    consumption_schedule = pd.Series(
+        data=power_values,
+        index=pd.date_range(
+            start=start,
+            end=pd.Timestamp(start) + pd.Timedelta(duration),
+            freq="15min",
+            inclusive="left",
+        ),
+    )
+
+    start_soc = message["flex-model"]["soc-at-start"] / 1000  # in MWh
+    roundtrip_efficiency = (
+        float(message["flex-model"]["roundtrip-efficiency"].replace("%", "")) / 100.0
+    )
+    storage_efficiency = (
+        float(message["flex-model"]["storage-efficiency"].replace("%", "")) / 100.0
+    )
+    soc_targets = message["flex-model"].get("soc-targets")
+
+    soc_schedule = integrate_time_series(
+        consumption_schedule,
+        start_soc,
+        up_efficiency=roundtrip_efficiency**0.5,
+        down_efficiency=roundtrip_efficiency**0.5,
+        storage_efficiency=storage_efficiency,
+        decimal_precision=6,
+    )
+    print(consumption_schedule)
+    print(soc_schedule)
+
+    # Check for absence of consumption breaches over 1 kW, i.e. any breach costs are avoided
+    assert all(v <= 0.001 for v in consumption_schedule)
+
+    # Check for absence of extra production peaks over 20 kW, i.e. any peak costs are avoided
```

Comment:
The `"site-peak-production"` (with its corresponding prices) is brought into action. The test fails if I comment out:
```
        "site-peak-consumption": "20 kW",
        "site-peak-production": "20 kW",
        "site-peak-consumption-price": "260 EUR/MW",
        "site-peak-production-price": "260 EUR/MW",
```
with:
```
# Check for absence of extra production peaks over 20 kW, i.e. any peak costs are avoided
>       assert all(v >= -0.02 for v in consumption_schedule)
```

The relaxation on the `"site-consumption-capacity"` isn't brought into action, though. If I comment out:
```
        "site-consumption-breach-price": "1000 EUR/kW",
        "site-production-breach-price": "1000 EUR/kW",
```
The capacity limit is treated as a hard constraint. Should I make an new Issue to reformulate the problem for this test so that it fails in case of a hard capacity constraint?

Reply:
Thanks. Yes, good idea about the hard capacity constraint. Turning breach pricing on or off could also be a parameter to this test.